### DISCUSSION TITLE
FavoriteButton uploade to mongoDB

### DISCUSSION THREAD

**Discussion by StephanLippert** [neuefische/questions#792] [Web] [answered]
Hi, we have a small problem with sending up our favorite button to MongoDB.
We have created a new model and used .populate in the other one (Recipe-model).
Now for the part we don't understand:
How do we send the FavoriteButton, WITHOUT an extra submitButton, to the database?
We are grateful for any help. 

https://github.com/gwittm/Digital-Recipe-Book/tree/US12-FavoritePage-Stephan-secondTry

**Comment by NikRoe** [comment, marked as answer]
Hi Stephan,

to send information to your database, you could have a simple button that has an onClick.
Whenever that button (most likely your favourite button) is clicked, you send a request to your API Route. This API Route then sends a request to your database.

Yesterday Pete and I gave the whole approach with a second model only for the favourites another thought and decided that it is a lot better to extend the documents (your single recipes) to include an isFavourite property.

Whenever your make the above mentioned call the the API Route, that API Route makes a request to your database to toggle the isFavourite property all your recipes then have.
